README.md

```diff
@@ -217,20 +228,21 @@ Although we think using Web IDL is probably not a good idea for JSON formats, as
 dictionary Scope {
   DOMString path_prefix;
   record<DOMString, Scope> scopes;
-  required record<DOMString, Package> packages;
+  required record<DOMString, Package | DOMString> packages;
 }
 
 dictionary Package {
   DOMString path;
-  required DOMString main;
+  DOMString main;
 }
 ```
 
 While parsing package name maps, we validate:
 - If non-strings show up where strings were expected, the map is invalid.
 - If non-objects show up where objects were expected, the map is invalid.
 - If the `"packages"` member is missing in a scope object, the map is invalid.
-- If the `"main"` member is missing in a package object, the map is invalid.
+- If any `"packages"` record keys start or end with a `/` separator, the map is invalid.
+- Any _string_ entries in the `"packages"` record are converted into package entries with a `"main"`.
```

Comment:
Should we make this more comprehensively in tune with the reference implementation? Or maybe just link to the appropriate functions in the reference implementation so we don't have to keep them in sync while we're still iterating?

Reply:
This is actually something that is done differently between the implementation and the spec, in that the implementation doesn't actually clone the package map object and change it as we read it, while a C++ implementation would likely be constructing the package map data structure during the validation phase.

So I'd be weary of linking to the implementation on this one, and would rather refine the specification notes as the source of truth.

In the name of improving and making this more comprehensive, what else should be added do you think?